lib/pwwka/transmitter.rb

```diff
@@ -23,10 +23,17 @@ class Transmitter
 
     DEFAULT_DELAY_BY_MS = 5000
 
+    attr_reader :app_manages_connector
     attr_reader :channel_connector
 
-    def initialize
-      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
+    def initialize channel_connector: nil
+      if channel_connector
+        @app_manages_connector = true
+        @channel_connector = channel_connector
+      else
+        @app_manages_connector = false
+        @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
```

Comment:
Should we externalize this connection-name-creator?   Whoever creates connections outside of this class should still be using this format exactly, so having a single place to do this would make that easier?

Reply:
I think at some point we should think about it but for the first things I want to try for this I'm going to copy it to the point where we use it.